### ISSUE TITLE
Regression of `SylowSubgroup`

### ISSUE THREAD

**Issue by frankluebeck** [gap-system/gap#1306]
### Observed behaviour
In GAP 4.7.6 the following commands
```
gap> g := AtlasGroup("Fi24'.2");;
gap> s := SylowSubgroup(g, 2);; 
```
need about 40 to 50 seconds.
In the current master branch the same commands need about 20 times longer (800-900 seconds).
(Tried this several times with different random seeds.)
### Expected behaviour
It should be faster or at least not slower in the master branch.

**Comment by fingolfin** [comment]
Confirmed. The regression already happens in v4.8.0

**Comment by fingolfin** [comment]
Using `git bisect` it seems that commit f062f1076815a01c44e45af5cd48fca50ab5fbc9 by @hulpke introduced the regression. By reverting it, with latest master the computation goes down to ~45 seconds for me (this is actually faster than GAP 4.7.6, which on my 5 years old laptop takes 75-90 seconds).

However, that commit adds an extra check, so this is not a simple matter of just reverting it... Anyway, I guess it's best to wait for comments from @hulpke now :-).

**Comment by fingolfin** [comment]
Ah, that commit is of course from the non-public history, so the commit SHA-1 does not resolve. It is visible for me because my local clone of the git repository has the old history grafted to it, so that I can find the cause of regressions like this (I think this is a great example why this would have been a desirable thing).

Anyway, to make it easier to see what's going on, here is the full commit:
```diff
commit f062f1076815a01c44e45af5cd48fca50ab5fbc9 (HEAD, refs/bisect/bad)
Author: Alexander Hulpke <hulpke@math.colostate.edu>
Date:   2014-06-26 15:04:40 -0600

    Better test for being blocks homomorphism.

diff --git a/lib/oprt.gi b/lib/oprt.gi
index 9bb39dfd0..4d0bb8469 100644
--- a/lib/oprt.gi
+++ b/lib/oprt.gi
@@ -674,7 +674,7 @@ local   xset,surj,G,  D,  act,  fam,  filter,  hom,  i;
         #fi;


-    # test for action on disjoint sets of numbers-> blocks homomorphism
+    # test for action on disjoint sets of numbers, preserved by group -> blocks homomorphism
     elif not IsExternalSubset( xset )
          and IsPermGroup( G )
          and IsList( D )
@@ -683,6 +683,8 @@ local   xset,surj,G,  D,  act,  fam,  filter,  hom,  i;
          and act = OnSets
     # disjointness test
     and Length(Set(Flat(D)))=Sum(List(D,Length))
+    # preserved test
+    and ForAll(D,b->ForAll(GeneratorsOfGroup(G),g->OnSets(b,g) in D))
     then
         filter := IsBlocksHomomorphism;
         hom.reps := [  ];
```

**Comment by hulpke** [comment]
@fingolfin 

> visible for me because my local clone of the git repository has the old history grafted to it, so that I can find the cause of regressions like this (I think this is a great example why this would have been a desirable thing).

So why don't we simply do it?

**Comment by hulpke** [comment]
The change was a required fix, as in some situation an action on sets was erroneously recognized as a block homomorphism, causing a wrong result. I'll see whether it can be streamlined.

**Comment by hulpke** [comment]
This is fixed in #1294

**Comment by fingolfin** [comment]
@hulpke Thank you!
